File: python/array.py
import time

NUMBER = 10 ** 8  # 100 million
my_array = [0.5] * NUMBER

total_sum = 0
def sum_array(array, start, end):
    total_sum = 0
    for i in range(start, end):
        total_sum += array[i]
    return total_sum

start = time.time()
print(sum_array(my_array, 0, len(my_array)))
end = time.time()

print("Took {:f}s".format(end - start))

File: rust/array.rs
use std::time::Instant;

const ARRAY_SIZE: usize = 1_000_000_000; // Adjust the size based on your system's capabilities

fn calculate_sum(data: &[f64], start: usize, end: usize) -> f64 {
    let mut sum = 0.0;
    for i in start..end {
        sum += data[i];
    }
    sum
}

fn main() {
    let data = vec![0.5; ARRAY_SIZE];

    let start = Instant::now();
    let sum = calculate_sum(&data, 0, ARRAY_SIZE);
    let end = Instant::now();

    println!("Took {:?}", end.duration_since(start));

    if (sum - (ARRAY_SIZE as f64 / 2.0)).abs() > f64::EPSILON {
        println!("Failed sum: {}", sum);
    } else {
        println!("Total Sum: {}", sum);
    }
}
